### PR TITLE
Add smart search fuctionality

### PR DESCRIPTION
**Pull request by sjauld**

### CODE REVIEW

File: abra-lib/client.go [hunk 1]
```diff
@@ -214,6 +214,81 @@ func NewWithGuid(guid string) (*Client, error) {
...
+			return string(result), numberTypeABN
+		}
+	}
+
+	return "", numberTypeNone
+}
+
+type charType int
+
+const (
```
Comment (jufemaiz) [inline abra-lib/client.go]: oh i like these :)

File: abra-lib/client.go [hunk 2]
```diff
@@ -214,6 +214,81 @@ func NewWithGuid(guid string) (*Client, error) {
...
+			return string(result), numberTypeACN
+		case 11:
+			return string(result), numberTypeABN
+		}
+	}
+
+	return "", numberTypeNone
+}
+
+type charType int
```
Comment (jufemaiz) [inline abra-lib/client.go]: is it better practice to group types/const/etc above?

**Review by jufemaiz — APPROVED** [review]

File: abra-lib/client.go [hunk 3]
```diff
@@ -214,6 +214,81 @@ func NewWithGuid(guid string) (*Client, error) {
...
+)
+
+func charCheck(i byte) charType {
+	if i == 32 { // space
+		return charCheckWhitespace
+	}
+	if i > 47 && i < 58 { // 0-9
+		return charCheckNumber
+	}
+	return charCheckNumber
```
Comment (sjauld) [inline abra-lib/client.go]: BUG 🤦‍♂

File: abra-lib/client.go [hunk 4]
```diff
@@ -248,14 +248,15 @@ func entityNumberFromString(text string) (string, numberType) {
 		result := []byte{text[i]}
 		// get numbers until they stop
 		i++
+	CheckLoop:
```
Comment (jufemaiz) [inline abra-lib/client.go]: What is this?

**Review by jufemaiz — APPROVED** [review]

File: abra-lib/client.go (hunk 2 above)
Comment (sjauld) [reply abra-lib/client.go]: I think usually you just group all of the functionality involving a type in the same area, so when you refactor you would just grab that whole section.

File: abra-lib/client.go (hunk 4 above)
Comment (sjauld) [reply abra-lib/client.go]: A break marker, see line 259.

**Pull request merged**